src/app/clusters/dishwasher-alarm-server/dishwasher-alarm-server.cpp

```diff
@@ -0,0 +1,299 @@
+    return status;
+}
+
+EmberAfStatus DishwasherAlarmServer::SetStateValue(EndpointId endpoint, BitMask<AlarmMap> newState)
+{
+    EmberAfStatus status = EMBER_ZCL_STATUS_SUCCESS;
+    BitMask<AlarmMap> currentState;
+
+    status = Attributes::State::Get(endpoint, &currentState);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        ChipLogProgress(Zcl, "Dishwasher Alarm: ERR: reading  state, err:0x%x", status);
+        return status;
+    }
+
+    status = Attributes::State::Set(endpoint, newState);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        ChipLogProgress(Zcl, "Dishwasher Alarm: ERR: writing  state, err:0x%x", status);
+        return status;
+    }
+
+    ChipLogProgress(Zcl, "Dishwasher Alarm: State ep%d value: %" PRIu32 "", endpoint, newState.Raw());
+
+    // Maintain consistency between latch and state states
+    BitMask<AlarmMap> latch;
+    status = GetLatchValue(endpoint, &latch);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        return status;
+    }
+
+    if (newState != latch)
+    {
+        latch.SetField(latch, newState.Raw());
+        status = SetLatchValue(endpoint, latch);
+    }
+
+    // Generate Notify event.
+    BitMask<AlarmMap> becameActive;
+    becameActive.Set(newState).Clear(currentState);
+    BitMask<AlarmMap> becameInactive;
+    becameInactive.Set(currentState).Clear(newState);
+    // This field SHALL be a copy of the Mask attribute when this event was generated.
+    BitMask<AlarmMap> mask;
+
+    status = GetMaskValue(endpoint, &mask);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        return status;
+    }
+    SendNotifyEvent(endpoint, becameActive, becameInactive, newState, mask);
+
+    return status;
+}
+
+void DishwasherAlarmServer::SendNotifyEvent(EndpointId endpointId, BitMask<AlarmMap> becameActive, BitMask<AlarmMap> becameInactive,
+                                            BitMask<AlarmMap> newState, BitMask<AlarmMap> mask)
+{
+    Events::Notify::Type event{ .active = becameActive, .inactive = becameInactive, .state = newState, .mask = mask };
+    EventNumber eventNumber;
+    CHIP_ERROR error = LogEvent(event, endpointId, eventNumber);
+    if (CHIP_NO_ERROR != error)
+    {
+        ChipLogError(Zcl, "[Notify] Unable to send notify event: %s [endpointId=%d]", error.AsString(), endpointId);
+    }
+}
+
+static Status resetHandler(const app::ConcreteCommandPath & commandPath, const chip::BitMask<AlarmMap> alarms)
+{
+    EndpointId endpoint = commandPath.mEndpointId;
+
+    EmberAfStatus status;
+    chip::BitMask<AlarmMap> state = 0;
+    status                        = State::Get(endpoint, &state);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        return Status::Failure;
+    }
+
+    chip::BitMask<AlarmMap> latch = 0;
+    status                        = Latch::Get(endpoint, &latch);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        return Status::Failure;
+    }
+
+    uint32_t alarmValue = alarms.Raw();
```

Comment:
Any use of Raw() outside logging is almost certainly wrong.  There shouldn't be any.